spec/controllers/admin/rentals_controller_spec.rb

```diff
@@ -97,6 +97,40 @@
       end
     end
 
+    context "when there is large amount of blank remote accounts" do
+      let!(:blank_remote_accounts) { create_list(:remote_account, 23, account: account) }
+
+      context "when configuration is set to not ignore blank remote accounts" do
+        it "assigns all remote accounts" do
+          index_with_search
+          expect(assigns(:remote_accounts).count).to eq(26)
+        end
+      end
+
+      context "when configuration is set to ignore blank remote accounts" do
+        around do |example|
+          BookingsyncPortal.ignore_blank_remote_accounts_for_account = ->(_) { true }
+          example.run
+          BookingsyncPortal.ignore_blank_remote_accounts_for_account = ->(_) { false }
+        end
+
+        it "ignores blank remote accounts on first page" do
+          index_with_search
+          expect(assigns(:remote_accounts).count).to eq(2)
+        end
+
+        context "when search" do
```

Comment:
brevity went too far for my liking :P 
```suggestion
        context "when there is rentals_search query" do
```